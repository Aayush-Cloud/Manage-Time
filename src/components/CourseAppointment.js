// @flow
import React from 'react';
import { View, Text } from 'react-native';
import styles from './styles/CourseAppointment.styles';

type Props = {
  name: string,
  starttime: moment$Moment,
  endtime: moment$Moment,
  location?: string,
  type?: string,
};

export default function CourseAppointment(props: Props) {
  const { starttime, endtime, name, location, type } = props;
  const start = starttime.format('H:mm');
  const end = endtime.format('H:mm');
  const secondaryInfo =
    (location || type) &&
    `${location || ''}${(location && type && ', ') || ''}${type || ''}`;
  return (
    <View style={styles.container}>
      <View style={styles.infoSpacer}>
        <View style={styles.timeContainer}>
          <Text style={[styles.time, styles.secondaryText]}>{start}</Text>
          <View style={styles.dash} />
          <Text style={[styles.time, styles.secondaryText]}>{end}</Text>
        </View>
      </View>
      <View style={styles.infoContainer}>
        <Text style={styles.title}>{name}</Text>
        {secondaryInfo && (
          <View style={styles.lineSpacer}>
            <Text style={styles.secondaryText}>{secondaryInfo}</Text>
          </View>
        )}
      </View>
    </View>
  );
}

CourseAppointment.defaultProps = {
  location: undefined,
  type: undefined,
};
