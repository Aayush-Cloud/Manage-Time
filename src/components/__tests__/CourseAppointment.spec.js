// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';
import CourseAppointment from '../CourseAppointment';

const fixedMoment = moment('01-01-2017', 'MM-DD-YYYY').startOf('day');

const props = {
  name: 'CourseAppointment name',
  starttime: fixedMoment,
  endtime: fixedMoment.clone().add(1, 'hour'),
  onEdit: jest.fn(),
};

describe('CourseAppointment component', () => {
  it('renders correctly without optional props', () => {
    const tree = renderer.create(<CourseAppointment {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with optional props', () => {
    const tree = renderer.create(
      <CourseAppointment {...props} location="Test location" type="Lecture" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
