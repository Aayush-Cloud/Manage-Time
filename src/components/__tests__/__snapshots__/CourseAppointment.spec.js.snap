// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseAppointment component renders correctly with optional props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 72,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
        "width": 72,
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
          "left": 16,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "Helvetica Neue",
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        0:00
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgba(0, 0, 0, 0.54)",
            "height": 1.5,
            "marginVertical": 4,
            "width": 5,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "Helvetica Neue",
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        1:00
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#00796B",
          "fontFamily": "Helvetica Neue",
          "fontSize": 20,
          "fontWeight": "400",
        }
      }
    >
      CourseAppointment name
    </Text>
    <View
      style={
        Object {
          "marginTop": 6,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "Helvetica Neue",
            "fontSize": 16,
            "fontWeight": "400",
          }
        }
      >
        Test location, Lecture
      </Text>
    </View>
  </View>
</View>
`;

exports[`CourseAppointment component renders correctly without optional props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 72,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
        "width": 72,
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
          "left": 16,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "Helvetica Neue",
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        0:00
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgba(0, 0, 0, 0.54)",
            "height": 1.5,
            "marginVertical": 4,
            "width": 5,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "Helvetica Neue",
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        1:00
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#00796B",
          "fontFamily": "Helvetica Neue",
          "fontSize": 20,
          "fontWeight": "400",
        }
      }
    >
      CourseAppointment name
    </Text>
  </View>
</View>
`;
